### PR TITLE
New map generator

### PR DESCRIPTION
**Pull request by valdisz**

### CODE REVIEW

File: aregion.cpp [hunk 1]
```diff
@@ -2743,6 +2748,718 @@ int mapBiome(int biome) {
...
+	for (auto wb : list) {
+		if (isNearWaterBody(reg, wb)) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+void makeRivers(Map* map, ARegionArray* arr, std::vector<WaterBody*>& waterBodies, std::unordered_map<ARegion*, int>& rivers,
```
Comment (artyomtrityak) [inline aregion.cpp:2835]: should it go to mapgen instead of region file?

File: neworigins/.gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+node_modules/
```
Comment (artyomtrityak) [inline neworigins/.gitignore:1]: we should delete all js/ts/json files

File: aregion.cpp (hunk 1 above)
Comment (valdisz) [reply aregion.cpp:2859]: then there will be a circular dependency

File: neworigins/.gitignore (hunk 2 above)
Comment (valdisz) [reply neworigins/.gitignore:1]: when this is done
I committed map rendering to allow others see the map w/o client while development happens

**Review by artyomtrityak — APPROVED** [review]

**Pull request merged**